tests/backup/backup_kdmp_test.go

```diff
@@ -0,0 +1,690 @@
+package tests
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/pborman/uuid"
+
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/sched-ops/k8s/storage"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/volume/portworx/schedops"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+	storagev1 "k8s.io/api/storage/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// This testcase verifies backup and restore with mentioned valid directories or files from backed-up apps and restores them when invalid,non-existend stoargeClass and files are there in KDMP exclude list
+var _ = Describe("{ExcludeInvalidDirectoryFileBackup}", func() {
+	var (
+		backupName                    string
+		scheduledAppContexts          []*scheduler.Context
+		AppContextsMapping            map[string]*scheduler.Context
+		namespace                     string
+		bkpNamespaces                 []string
+		backupNames                   []string
+		restoreNames                  []string
+		scheduleNames                 []string
+		clusterUid                    string
+		clusterStatus                 api.ClusterInfo_StatusInfo_Status
+		restoreName                   string
+		cloudCredName                 string
+		cloudCredUID                  string
+		backupLocationUID             string
+		bkpLocationName               string
+		numDeployments                int
+		providers                     []string
+		backupLocationMap             map[string]string
+		labelSelectors                map[string]string
+		storageClassExcludeFileDirMap map[*storagev1.StorageClass][]string
+		mountPathExcludeFileDirMap    map[string][]string
+		existingFileDirMountPathMap   map[string][]string
+		fileListMountMap              map[string][]string
+		dirListMountMap               map[string][]string
+		finalFileList                 map[string][]string
+		scMountPathMap                map[string]*storagev1.StorageClass
+		backupNamespaceMap            map[string]string
+		podScMountPathMap             map[string]map[string]*storagev1.StorageClass
+		excludeList                   string
+		fileList                      []string
+		dirList                       []string
+		preRuleName                   string
+		postRuleName                  string
+		preRuleUid                    string
+		postRuleUid                   string
+		periodicSchedulePolicyName    string
+		periodicSchedulePolicyUid     string
+		mutex                         sync.Mutex
+		restoredNamespaces            []string
+		wg                            sync.WaitGroup
+	)
+	JustBeforeEach(func() {
+		backupNames = make([]string, 0)
```

Comment:
Why can't we put it in the var declaration  above?

Reply:
Thanks @kshithijiyer-px for suggestion, Corrected it.